_rules/sequentially-focusable-element-has-visible-focus-oj04fd.md

```diff
@@ -0,0 +1,181 @@
+
+### Passed
+
+#### Passed Example 1
+
+The first [focusable][] element is part of [sequential focus navigation][]. It has an outline when it is [focused][] (due to default User Agent's styling). The second [focusable][] element is not applicable because it has been removed from [sequential focus navigation][] due to the negative value for the `tabindex` attribute. Its presence is nonetheless enough to make the first one applicable.
+
+```html
+<a href="https://act-rules.github.io/">ACT rules</a> <button tabindex="-1">Dummy button</button>
+```
+
+#### Passed Example 2
+
+The first [focusable][] element is part of [sequential focus navigation][]. It has an outline when it is [focused][] (due to default User Agent's styling).
+
+```html
+<span tabindex="0">Act rules</span> <button tabindex="-1">Dummy button</button>
+```
+
+#### Passed Example 3
+
+The first [focusable][] element, is part of [sequential focus navigation][]. Its parent `span` puts a border around it when it is [focused][] (the border of the `span` element).
+
+```html
+<link rel="stylesheet" href="../test-assets/focus-visible/styles.css" />
+<script src="../test-assets/focus-visible/script.js"></script>
+
+<span id="indicator" class="border">
+	<a
+		id="act"
+		class="no-focus-default"
+		onfocus="toggleActivation('indicator')"
+		onblur="toggleActivation('indicator')"
+		href="https://act-rules.github.io/"
+		>ACT rules</a
+	>
+</span>
+<button tabindex="-1">Dummy button</button>
+```
+
+#### Passed Example 4
+
+Each of these three [focusable][] elements has a blue square in front of it when it is [focused][].
+
+```html
+<link rel="stylesheet" href="../test-assets/focus-visible/styles.css" />
+<script src="../test-assets/focus-visible/script.js"></script>
+
+<span id="indicator-act" class="indicator solid"></span>
+<a
+	id="act"
+	class="no-focus-default"
+	onfocus="toggleActivation('indicator-act'); toggleActivation('indicator-wcag')"
+	onblur="toggleActivation('indicator-act'); toggleActivation('indicator-wcag')"
+	href="https://act-rules.github.io/"
+	>ACT rules</a
+>
+<span id="indicator-wcag" class="indicator solid"></span>
+<a
+	id="wcag"
+	class="no-focus-default"
+	onfocus="toggleActivation('indicator-wcag'); toggleActivation('indicator-w3c')"
+	onblur="toggleActivation('indicator-wcag'); toggleActivation('indicator-w3c')"
+	href="https://www.w3.org/TR/WCAG21/"
+	>WCAG</a
+>
+<span id="indicator-w3c" class="indicator solid"></span>
+<a
+	id="w3c"
+	class="no-focus-default"
+	onfocus="toggleActivation('indicator-w3c'); toggleActivation('indicator-final')"
+	onblur="toggleActivation('indicator-w3c'); toggleActivation('indicator-final')"
+	href="https://www.w3.org/"
+	>WCAG</a
+>
+<span id="indicator-final" class="indicator solid"></span>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The first [focusable][] element does not have any pixel changing color when it is [focused][] because the default styling has been overwritten by a style that removes the outline.
```

Comment:
I would place the button in the sequential focus order (i.e. remove the negative tabindex) to make it clear that even if one of the elements pass the rule, it fails because all need to pass it

Reply:
I'm afraid that mixing Passed and Failed targets in the same test case will be messing up the implementation reports.
Currently, we expect one "failed" result for this test case. If we make the other one passing, we'd expect one "fail" and one "passed" result, and thus we also need to figure out which outcome is for which target in order to decide if the implementation is correct, and that makes automated implementations mapping harder, I guess…

Reply:
That's fair